File: src/main/java/org/angry1980/security/jwt/JwtAuthenticationToken.java
package org.angry1980.security.jwt;

import io.jsonwebtoken.Claims;
import io.jsonwebtoken.Jws;
import org.springframework.security.authentication.AbstractAuthenticationToken;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.userdetails.User;

import java.util.Collection;
import java.util.Objects;

public class JwtAuthenticationToken extends AbstractAuthenticationToken {

    private final Jws<Claims> token;
    private final User user;

    public JwtAuthenticationToken(Jws<Claims> token,
                                  User user,
                                  Collection<? extends GrantedAuthority> authorities) {
        super(authorities);
        this.token = Objects.requireNonNull(token);
        this.user = Objects.requireNonNull(user);
        this.setAuthenticated(true);

    }

    public JwtAuthenticationToken(Jws<Claims> token) {
        super(null);
        this.token = Objects.requireNonNull(token);
        this.user = null;
        this.setAuthenticated(false);
    }

    public Object getCredentials() {
        return token;
    }

    public Object getPrincipal() {
        return user;
    }
}

File: src/main/java/org/angry1980/security/jwt/JwtAuthenticationToken.kt
package org.angry1980.security.jwt

import io.jsonwebtoken.Claims
import io.jsonwebtoken.Jws
import org.springframework.security.authentication.AbstractAuthenticationToken
import org.springframework.security.core.GrantedAuthority
import org.springframework.security.core.userdetails.User
import java.util.Objects

/**
 * Wrapper for signed json web token compatible with spring security solution
 */
class JwtAuthenticationToken(
        val token: Jws<Claims>,
        val user: User? = null,
        authorities: Collection<GrantedAuthority>? = null
) : AbstractAuthenticationToken(authorities) {

    init {
        if(user != null){
            isAuthenticated = true
        }
    }

    override fun getCredentials() = token

    override fun getPrincipal() = user

}
